### PR TITLE
[FEATURE] [BREAKING] Support writing logs in a file.

### LINKED ISSUE #338: Be able to save logs in file
Currently Authelia logs are not very good as there is a lot of redundant information and the format is not very user-friendly.

### PR DESCRIPTION
**Pull request by clems4ever**
Fixes #338

### CODE REVIEW

File: docs/configuration/miscellaneous.md [hunk 1]
```diff
@@ -29,7 +29,9 @@ following configuration options:
     tls_cert: /var/lib/authelia/ssl/cert.pem
 
 
-## Logs level
+## Logs
```
Comment (nightah) [inline docs/configuration/miscellaneous.md]: I think we should consider changing references from `logs` to `log`.

```suggestion
## Log
```

File: config.template.yml [hunk 2]
```diff
@@ -10,6 +10,8 @@ port: 9091
 
 # Level of verbosity for logs: info, debug, trace
 logs_level: debug
```
Comment (nightah) [inline config.template.yml]:
```suggestion
log_level: debug
```

File: config.template.yml [hunk 3]
```diff
@@ -10,6 +10,8 @@ port: 9091
...
+## File path where the logs will be written. If not set logs are written to stdout.
+# logs_file_path: /var/log/authelia
```
Comment (nightah) [inline config.template.yml]:
```suggestion
# log_file_path: /var/log/authelia
```

File: docs/configuration/miscellaneous.md [hunk 4]
```diff
@@ -29,7 +29,9 @@ following configuration options:
...
+
+### Logs level
```
Comment (nightah) [inline docs/configuration/miscellaneous.md]:
```suggestion
### Log level
```

File: docs/configuration/miscellaneous.md [hunk 5]
```diff
@@ -38,6 +40,15 @@ Defines the level of logs used by Authelia. This level can be set to
 
     logs_level: debug
```
Comment (nightah) [inline docs/configuration/miscellaneous.md]:
```suggestion
    log_level: debug
```

File: docs/configuration/miscellaneous.md [hunk 6]
```diff
@@ -38,6 +40,15 @@ Defines the level of logs used by Authelia. This level can be set to
...
 
+### Logs file path
```
Comment (nightah) [inline docs/configuration/miscellaneous.md]:
```suggestion
### Log file path
```

File: docs/configuration/miscellaneous.md [hunk 7]
```diff
@@ -38,6 +40,15 @@ Defines the level of logs used by Authelia. This level can be set to
...
+
+`optional: true`
+
+Logs can be stored in a file when file path is provided. Otherwise logs
+are written to standard output.
+
+    logs_file_path: /var/log/authelia.log
```
Comment (nightah) [inline docs/configuration/miscellaneous.md]:
```suggestion
    log_file_path: /var/log/authelia.log
```

File: internal/configuration/schema/configuration.go [hunk 8]
```diff
@@ -7,7 +7,8 @@ type Configuration struct {
 	TLSCert string `mapstructure:"tls_cert"`
 	TLSKey  string `mapstructure:"tls_key"`
 
-	LogsLevel string `mapstructure:"logs_level"`
+	LogsLevel    string `mapstructure:"logs_level"`
```
Comment (nightah) [inline internal/configuration/schema/configuration.go]:
```suggestion
	LogLevel    string `mapstructure:"logs_level"`
```

File: internal/configuration/schema/configuration.go [hunk 9]
```diff
@@ -7,7 +7,8 @@ type Configuration struct {
...
+	LogsFilePath string `mapstructure:"logs_file_path"`
```
Comment (nightah) [inline internal/configuration/schema/configuration.go]:
```suggestion
	LogFilePath string `mapstructure:"log_file_path"`
```

File: cmd/authelia/main.go [hunk 10]
```diff
@@ -42,6 +38,10 @@ func startServer() {
 		panic(errors.New("Some errors have been reported"))
 	}
 
+	if err := logging.InitializeLogger(config.LogsFilePath); err != nil {
```
Comment (nightah) [inline cmd/authelia/main.go]:
```suggestion
	if err := logging.InitializeLogger(config.LogFilePath); err != nil {
```

File: cmd/authelia/main.go [hunk 11]
```diff
@@ -42,6 +38,10 @@ func startServer() {
...
+		log.Fatalf("Cannot initialize logger: %v", err)
+	}
+
 	switch config.LogsLevel {
```
Comment (nightah) [inline cmd/authelia/main.go]:
```suggestion
	switch config.LogLevel {
```

**Comment by clems4ever** [comment]
@nightah , will do but it will be a breaking change. I'll advertise it in BREAKING.md and in the commit message with a [BREAKING] tag.

File: BREAKING.md [hunk 12]
```diff
@@ -6,6 +6,10 @@ recommended not to use the 'latest' Docker image tag blindly but pick a version
 and read this documentation before upgrading. This is where you will get information about
 breaking changes and about what you should do to overcome those changes.
 
+## Breaking in v4.7.0
+
+`logs_level` configuration is renamed `log_level`.
```
Comment (nightah) [inline BREAKING.md]:
```suggestion
`logs_level` configuration key has been renamed to `log_level`.
```

**Pull request merged**